### DISCUSSION TITLE
Output standalone mood not omtimizing images?

### DISCUSSION THREAD

**Discussion by siamahnaf** [vercel/next.js#41129] [Help]
Hello, I use [automatically coping traced file](https://nextjs.org/docs/advanced-features/output-file-tracing#automatically-copying-traced-files) for deploying my site to ec2 server. I follow this documentation-

https://nextjs.org/docs/advanced-features/output-file-tracing#automatically-copying-traced-files

Here first of all I updated my next.config.js file-

```
module.exports = {
  output: 'standalone',
}
```

Then after building I manually copy `static` folder to `standalone/.next/` and also copy `public` folder to `standalone/`. After copying I run `node server.js`. It is running successfully. But image optimization not working.

In `terminal` I found this error-

`Error: 'sharp' is required to be installed in standalone mode for the image optimization to function correctly`

But When I check `standalone/package.json` and `standalone/node_modules` I can see It already have sharp package.

Then after some researching, I also add a environment variable to `standalone/.env` file as `NEXT_SHARP_PATH=/tmp/node_modules/sharp next start`

Image optimization still not working.

I try to some different way -
`NEXT_SHARP_PATH=/tmp/node_modules/sharp next standalone`,
`NEXT_SHARP_PATH=/tmp/node_modules/sharp standalone`,
`NEXT_SHARP_PATH=/node_modules/sharp`,
`NEXT_SHARP_PATH=/tmp/node_modules/sharp`

***Note: I use linux subsystem on windows. In linux subsystem, When I run product mode `npm run start`. Image optmization automatically work in this mode, I haven't to give `.env` also. When I start my project in standalone mood, image optimization is not working. Even when I give `.env` it's not working.

Here is my `next.config.js`-

```
/** @type {import('next').NextConfig} */
const withPWA = require("next-pwa")({
  dest: "public",
  register: true,
  disable: process.env.NODE_ENV === "development"
});

const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  output: "standalone",
  images: {
    minimumCacheTTL: 2592000,
    domains: ["dzzoo94ozikw5.cloudfront.net", "lh3.googleusercontent.com", "platform-lookaside.fbsbx.com"]
  },
}

module.exports = withPWA(nextConfig)
```

Still image optimization not working. How can I fix that problem. Please help me. I am stacking with this about 30 days. Please please please.

**Comment by harrybawsac** [comment]
I have exactly the same issue. I tried everything!

- All sorts of values for `NEXT_SHARP_PATH` pointing to all sorts of places or not using `NEXT_SHARP_PATH` at all. 
- I tried both NPM and Yarn.
- Installed v0.33.2 and v0.32.6 of Sharp, as some suggested.
- Installed platform specific [versions of Sharp](https://sharp.pixelplumbing.com/install#npm-v10).
- Running the correct server.js file.

But I keep getting `Error: 'sharp' is required to be installed in standalone mode for the image optimization to function correctly. Read more at: https://nextjs.org/docs/messages/sharp-missing-in-production`.

Please help.

I'm running Ubuntu 22. No Docker stuff or anything.

**Comment by siamahnaf** [comment]
You need to build from linux or macos, then you can try. I believe the problem will be solved.

**Reply by harrybawsac** [reply]
Yeah I tried Linux. [But I managed to fix it!](https://github.com/vercel/next.js/discussions/41129#discussioncomment-8382033)

**Comment by harrybawsac** [comment]
I just managed to fix it!

Removing everything `NEXT_SHARP_PATH` related. Downgrade sharp to v0.32.6. Suddenly it worked. Then Next sees the node module automatically.

**Reply by axel-ojje** [reply]
^ It worked for me when deploying to GCP. I also removed `package-lock.json,` reinstalled it, and then deployed via Google CLI. No more telemetry warning.

**Reply by sgoodrow-zymergen** [reply]
Yes, this was the issue. Don't use latest. Sigh. I really hope that an image optimization library is included in turbopack's builder so this dependency can be omitted.
